cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageDataset.java

```diff
@@ -157,9 +162,9 @@ public void addAccess(Id.Run run, Id.Stream stream,
           LOG.trace("Got row key = {}", Bytes.toString(row.getRow()));
         }
         RowKey rowKey = parseRow(row);
-        if (run.getId().equals(rowKey.getRunId().getId())) {
-          recordBuilder.add(rowKey.getProgram().toEntityId());
-          recordBuilder.add((NamespacedEntityId) rowKey.getData().toEntityId());
+        if (run.getEntityName().equals(rowKey.getRunId().getId())) {
```

Comment:
@bdmogal it currently returns a `RunId` from `org.apache.twill.api.RunId` should it be changed to return `ProgramRunId` ?